tests/helpers/test_intent.py

```diff
@@ -158,15 +158,41 @@ def test_async_validate_slots() -> None:
     )
 
 
-async def test_cant_turn_on_sun(hass: HomeAssistant) -> None:
-    """Test we can't turn on entities that don't support it."""
+async def test_cant_turn_on_sensor(hass: HomeAssistant) -> None:
+    """Test that we can't turn on entities that don't support it."""
     assert await async_setup_component(hass, "homeassistant", {})
     assert await async_setup_component(hass, "conversation", {})
     assert await async_setup_component(hass, "intent", {})
-    assert await async_setup_component(hass, "sun", {})
+    assert await async_setup_component(hass, "sensor", {})
+
+    hass.states.async_set(
+        "sensor.test", "123", attributes={ATTR_FRIENDLY_NAME: "Test Sensor"}
+    )
+
     result = await conversation.async_converse(
-        hass, "turn on sun", None, Context(), None
+        hass, "turn on test sensor", None, Context(), None
     )
 
     assert result.response.response_type == intent.IntentResponseType.ERROR
     assert result.response.error_code == intent.IntentResponseErrorCode.FAILED_TO_HANDLE
+
+
+async def test_exposed_domains(hass: HomeAssistant) -> None:
```

Comment:
Please add an area -> device -> entity relationship so we can test that. 

Also, this is a test for the default agent, so please move the test to that file.

Reply:
Moved test to default agent tests and added a new one checking devices/areas